packages/modal/src/modal.ts

```diff
@@ -36,22 +35,51 @@ class Modal {
         const { showModal, hideModal, autoInitModal, isHidden } = initModal(modalElement, triggerButton, this.options);
 
         if (this.state === "open") {
-            showModal()
+            this.playOpenAnimation(); // Use animation when opening the modal
```

Comment:
> As I mentioned earlier, the animation shouldn’t be applied at this level. It’s better to give users the flexibility to handle it themselves. Right now, if you’ve noticed, to animate the modal, users must provide keyframes (either through options or attributes) [(check here](https://flexilla-docs.vercel.app/docs/components/modal#animatecontent) Instead of requiring users to create keyframes each time and then pass them in as animations, the idea is to offer ready-made animations. So, if users want to use pre-configured animations, they should have the option to easily enable them. Does that make things clearer?

Ah, I see what you mean! So, we would provide a set of pre-configured animations, possibly through an enum or schema for different animation types, and then users can choose which animation they want to activate. This way, they don’t need to define their own keyframes every time, but they still have the flexibility to select from our predefined options.

Or is it something different that you're referring to?

Reply:
@shreyannandanwar, will you be taking on the improvements, or would you like some help with it?

Reply:
> Ah, I see what you mean! So, we would provide a set of pre-configured animations, possibly through an enum or schema for different animation types, and then users can choose which animation they want to activate. This way, they don’t need to define their own keyframes every time, but they still have the flexibility to select from our predefined options.

Hey @Adebesin-Cell you're right